subsys/usb/usb_c/usbc_tc_common.c

```diff
@@ -118,7 +123,11 @@ static void tc_init(const struct device *dev)
 	tc->flags = ATOMIC_INIT(0);
 
 	/* Initialize the TCPC */
-	tcpc_init(tcpc);
+	ret = tcpc_init(tcpc);
+	if (ret < 0) {
```

Comment:
Knowing the context why you write it this way and not `if (ret != 0)` or `if (ret)`, you introduce undefined behavior because positive return value for `tcpc_init();` is not defined and only 0 is defined as success.

Reply:
Hmm, it's really nice point here. To be honest, I used the comparison "< 0" because I found many functions that also had a negative error codes and 0 on success and their result were checked that way. I will change that to be more defensive

Reply:
Done